opal/mca/common/ofi/common_ofi.c

```diff
@@ -675,8 +677,31 @@ static bool is_near(pmix_device_distance_t *distances,
 
     return false;
 }
-#endif
-#endif  // OPAL_OFI_PCI_DATA_AVAILABLE
+#endif /* HWLOC_API_VERSION < 0x00020000 */
+
+static bool is_near(pmix_device_distance_t *distances,
+                    int num_distances,
+                    struct fi_info *fi_info)
+{
+    /* if we failed to find any distances, then we consider all interfaces
+     * to be of equal distances and let the caller decide how to handle
+     * them
+     */
+    if (!distances)
+        return true;
+
+    if (fi_info && fi_info->fabric_attr && fi_info->fabric_attr->prov_name
+        && 0 == strcasecmp("efa", fi_info->fabric_attr->prov_name)) {
```

Comment:
TBH I don't know. I would say case insensitive is better - it's less restrictive.

Reply:
nit: I  think this should be lower-case only.  It has been [since first introduced in 1.8.x](https://github.com/ofiwg/libfabric/blob/v1.8.x/prov/efa/src/efa.h#L67) and [continues](https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/efa_prov.c#L90) to be lower case.

Reply:
After thinking about `OPAL_OFI_PCI_DATA_AVAILABLE` a little bit, it seems `is_near` only needs that macro when calling `match_device_by_pci_bus_id`.  Do we need that macro if we want to use match_device_by_name_prefix for EFA?